(dp1
S'output'
p2
S"<type 'exceptions.AttributeError'> 'NoneType' object has no attribute 'username'"
p3
sS'layer'
p4
S'C:\\Users\\hd\\PycharmProjects\\untitled\\web2py\\applications\\darkcloud1\\controllers/default.py'
p5
sS'code'
p6
S'<code object <module> at 000000000A8136B0, file "C:\\Users\\hd\\PycharmProjects\\untitled\\web2py\\applications\\darkcloud1\\controllers/default.py", line 4>'
p7
sS'snapshot'
p8
(dp9
sS'traceback'
p10
S'Traceback (most recent call last):\n  File "C:\\Users\\hd\\PycharmProjects\\untitled\\web2py\\gluon\\restricted.py", line 219, in restricted\n    exec(ccode, environment)\n  File "C:\\Users\\hd\\PycharmProjects\\untitled\\web2py\\applications\\darkcloud1\\controllers/default.py", line 239, in <module>\n  File "C:\\Users\\hd\\PycharmProjects\\untitled\\web2py\\gluon\\globals.py", line 419, in <lambda>\n    self._caller = lambda f: f()\n  File "C:\\Users\\hd\\PycharmProjects\\untitled\\web2py\\applications\\darkcloud1\\controllers/default.py", line 198, in subscriber\n    return dict(usertype=type, user=session.user.username, devicestemp=devicestemp, form=addform)\nAttributeError: \'NoneType\' object has no attribute \'username\'\n'
p11
s.